decompiler/frontend/binaryninja/frontend.py

```diff
@@ -123,30 +53,28 @@ def from_raw(cls, view: BinaryView):
         """Create a binaryninja frontend instance based on an initialized binary view."""
         return cls(view)
 
-    def create_task(self, function_identifier: Union[str, Function], options: Options) -> DecompilerTask:
-        """Create a task from the given function identifier."""
-        function = FunctionObject.get(self._bv, function_identifier)
-        tagging = CompilerIdiomsTagging(self._bv, function.function.start, options)
-        tagging.run()
+    def lift(self, task: DecompilerTask):
+        if task.failed:
+            return
+
         try:
-            cfg, complex_types = self._extract_cfg(function.function, options)
-            task = DecompilerTask(
-                function.name,
-                cfg,
-                function_return_type=function.return_type,
-                function_parameters=function.params,
-                options=options,
-                complex_types=complex_types,
-            )
+            function = self._get_binninja_function(task.function_identifier)
+            lifter, parser = self._create_lifter_parser(task.options)
+
+            task.function_return_type = lifter.lift(function.return_type)
+            task.function_parameters = [lifter.lift(param_type) for param_type in function.type.parameters]
+
+            tagging = CompilerIdiomsTagging(self._bv, function.start, task.options)
+            tagging.run()
```

Comment:
What happens if the tagging crashes at some point? Should this also get a try an except?

Reply:
Previously any error from tagging would just bubble up all the way to the decompile method.
Now, it is caught by the try ... except in the lift method and causes the task to fail with "Function lifting".